python/cvmfs/repository.py

```diff
@@ -292,10 +459,9 @@ def retrieve_certificate(self):
         return Certificate(certificate)
 
 
-    @abc.abstractmethod
     def retrieve_file(self, file_name):
-        """ Abstract method to retrieve a file from the repository """
-        pass
+        """ Method to retrieve a file from the repository """
```

Comment:
Well, `retrieve_file()` downloads any URL without doing anything with it and shouldn't be exposed per-se. For downloading the manifest and whitelist we have specialised methods. On the other hand `retrieve_object()` gets and decompresses CAS objects in **data/../........** which might be useful indeed. Hence, I would just remove `retrieve_file()` from the public interface and keep `retrieve_object()` as is.